src/query/ast/src/parser/common.rs

```diff
@@ -97,12 +97,13 @@ pub fn function_name(i: Input) -> IResult<Identifier> {
 }
 
 pub fn stage_name(i: Input) -> IResult<Identifier> {
+    let named_stage = rule! { #plain_identifier(|token| token.is_reserved_ident(false)) };
```

Comment:
maybe extract `plain_identifier(is_reserved_keyword: fn(&TokenKind) -> bool)` to `plain_identifier(after_as: bool)` since it's commonly used.

Reply:
done